shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/constant/MySQLAuthenticationPlugin.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.db.protocol.mysql.constant;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * MySQL client/server protocol authentication plugins.
+ *
+ * @see <a href="https://dev.mysql.com/doc/refman/8.0/en/authentication-plugins.html">Authentication Plugins</a>
+ */
+@RequiredArgsConstructor
+@Getter
+public enum MySQLAuthenticationPlugin {
+    
+    DEFAULT(null),
+    
+    NATIVE_PASSWORD_AUTHENTICATION("mysql_native_password"),
+    
+    SHA2_AUTHENTICATION("caching_sha2_password"),
+    
+    SHA256_AUTHENTICATION("sha256_password");
+    
```

Comment:
1, From the doc, looks there's `mysql_clear_password`, `authentication_windows_client`, etc, is it enough?

2, Except `Client-side plugin`, looks there's also `Server-side plugin`, it has other options. Basd on MySQLAuthenticationPlugin javadoc, seems it includes server-side authentication plugins.

Reply:
https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_default_authentication_plugin